chain-signatures/contract/tests/tests.rs

```diff
@@ -371,3 +396,144 @@ async fn test_contract_initialization() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_contract_propose_update() {
+    let (_, contract, accounts, _) = init_env().await;
+    dbg!(contract.id());
+
+    test_propose_update_config(&contract, &accounts).await;
+    test_propose_update_contract(&contract, &accounts).await;
+}
+
+async fn test_propose_update_config(contract: &Contract, accounts: &[Account]) {
+    // contract should not be able to propose updates unless its apart of the participant/voter set.
+    let execution = contract
+        .call("propose_update")
+        .args_json(serde_json::json!({
+            "contract": vec![1, 2, 3],
+        }))
+        .transact()
+        .await
+        .unwrap();
+    dbg!(&execution);
+    assert!(execution.is_failure());
+
+    // have each participant propose a new update:
+    let new_config = Config {
+        triple_timeout: min_to_ms(10),
+        ..Default::default()
+    };
+    let mut proposals = Vec::with_capacity(accounts.len());
+    for account in accounts {
```

Comment:
I assume all of the participants will get different IDs even if the proposal is the same.
That is fine with me, just trying to understand the logic here.

Reply:
yep, not trying to track conflicting proposals. Just going to generate a new id per proposal